scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/IcebergEncoder.java

```diff
@@ -69,9 +104,29 @@ public <T> byte[] encode(T value, Coder<T> coder) {
         if (value instanceof LocalTime) {
             return encode((LocalTime) value);
         }
+        if (value instanceof LocalDateTime) {
+            return encode((LocalDateTime) value);
+        }
+        if (value instanceof ZonedDateTime) {
+            return encode((ZonedDateTime) value);
+        }
         if (value instanceof Instant) {
             return encode((Instant) value);
         }
+        if (value instanceof BigDecimal) {
+            return encode((BigDecimal) value);
+        }
+
+        // types below are not currently valid for bucketing but with a defined hash function
+        if (value instanceof Boolean) {
+            return encode((Boolean) value);
+        }
+        if (value instanceof Float) {
+            return encode((Float) value);
+        }
+        if (value instanceof Double) {
+            return encode((Double) value);
+        }
```

Comment:
I didn't add this because it's better to fail since they are not supported for bucket